crates/sncast/src/response/print.rs

```diff
@@ -0,0 +1,242 @@
+use anyhow::Result;
+use serde_json::Value;
+use starknet::core::types::FieldElement;
+use std::{collections::HashMap, fmt::Display, str::FromStr};
+
+use serde::{Serialize, Serializer};
+
+use crate::NumbersFormat;
+
+use super::structs::CommandResponse;
+
+pub enum OutputFormat {
+    Json,
+    Human,
+}
+
+impl OutputFormat {
+    #[must_use]
+    pub fn from_flag(json: bool) -> Self {
+        if json {
+            OutputFormat::Json
+        } else {
+            OutputFormat::Human
+        }
+    }
+}
+
+/// Constrained subset of `serde::json`
+#[derive(PartialEq, Eq, Debug)]
+pub enum OutputValue {
+    String(String),
+    Array(Vec<OutputValue>),
+}
+
+type OutputData = Vec<(String, OutputValue)>;
```

Comment:
Why not add `OrderedMap(IndexMap<String, OutputValue>)`? Then `OutputData` will not be needed

Reply:
Beacuse we allow maps only on the top level.

Reply:
Added comment about this